src/main/java/io/javalin/core/JavalinServlet.kt

```diff
@@ -74,7 +77,25 @@ class JavalinServlet(
                 }
             }
             if (endpointEntries.isEmpty() && type != HandlerType.HEAD && type != HandlerType.GET) {
-                throw HaltException(404, "Not found")
+
+                if (prefer405over404) {
```

Comment:
The body of the condition can be moved to the private function to prevent bloating the method with logic.